test/test-functions

```diff
@@ -1421,16 +1425,16 @@ do_test() {
         [[ -d $usrlibdir ]] && libdirs+=" $usrlibdir" && break
     done
 
+    mkdir -p "$STATEDIR"
+
     import_testdir
     import_initdir
 
     while (($# > 0)); do
         case $1 in
             --run)
                 echo "TEST RUN: $TEST_DESCRIPTION"
-                test_run
-                ret=$?
```

Comment:
`ret=$?` was removed, but the code still runs `exit $ret`. I wonder what the value of `$ret` is.